src/Module/Notifications/Notification.php

```diff
@@ -107,24 +108,31 @@ public static function content(array $parameters = [])
 		$request_id = $parameters['id'] ?? false;
 
 		if ($request_id) {
-			$notify = DI::notify()->getByID($request_id, local_user());
+			$Notify = DI::notify()->selectOneById($request_id);
+			if ($Notify->uid !== local_user()) {
+				throw new HTTPException\ForbiddenException();
+			}
 
 			if (DI::pConfig()->get(local_user(), 'system', 'detailed_notif')) {
-				$notify->seen = true;
-				DI::notify()->update($notify);
+				$Notify->setSeen();
+				DI::notify()->save($Notify);
 			} else {
-				DI::notify()->setSeen(true, $notify);
+				if ($Notify->uriId) {
+					DI::notification()->setAllSeenForUser($Notify->uid, ['target-uri-id' => $Notify->uriId]);
+				}
+
+				DI::notify()->setAllSeenForRelatedNotify($Notify);
```

Comment:
Same point as above: Maybe we do need a "transactional" behavior for such cases?